saas_public/controllers/saas_public.py

```diff
@@ -5,12 +5,14 @@
 
 
 class SaaSPublicController(Controller):
-    @route('/saas_public/<model("saas.template"):template_id>/create-fast-build', type='http', auth='public')
+    @route('/saas_public/<int:template_id>/create-fast-build', type='http', auth='public')
     def create_fast_build(self, template_id, **kwargs):
         if not kwargs:
             kwargs = {}
-        if template_id and template_id.sudo().public_access:
-            template_operator_id = template_id.sudo().operator_ids.random_ready_operator()
+        if template_id:
+            template = request.env['saas.template'].browse(template_id)
+        if template.sudo().public_access:
```

Comment:
You either don't check that ``template_id`` is specified or you check that template variable is set. I prefer the first option